TST use global_random_seed in sklearn/utils/tests/test_stats.py

Rishab260 opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
towards https://github.com/scikit-learn/scikit-learn/issues/22827

#### What does this implement/fix? Explain your changes.
I added the global_random_seed fixture to the following tests:-

- test_weighted_percentile_2d
- test_weighted_median_equal_weights
- test_averaged_weighted_percentile
- test_weighted_median_integer_weights

#### Any other comments?
Feedback and guidance is highly appreciated. Looking forward to contributing to this project more and more in the coming days. Thanks in advance!

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

Rishab260 committed:
add global_random_seed to test_averaged_weighted_percentile

Rishab260 committed:
add global_random_seed to test_weighted_median_equal_weights

Rishab260 committed:
add global_random_seed to test_weighted_median_integer_weights

Rishab260 committed:
add global_random_seed to test_weighted_percentile_2d

Rishab260 committed:
test_stats  [all random seeds]
test_averaged_weighted_percentile
test_weighted_median_equal_weights
test_weighted_median_integer_weights
test_weighted_percentile_2d

Rishab260 committed:
Merge branch 'main' into test_stats_branch

Rishab260 committed:
Merge branch 'main' into test_stats_branch

Rishab260 committed:
Merge branch 'main' into test_stats_branch

OmarManzoor reviewed (commented):
Thanks for the PR @Rishab260 

Please also add a commit with all global seeds for the respective tests on this PR too.

Rishab260 committed:
test_stats [all random seeds]
test_averaged_weighted_percentile
test_weighted_median_equal_weights
test_weighted_median_integer_weights
test_weighted_percentile_2d

Rishab260 committed:
Merge branch 'main' into test_stats_branch

Rishab260 committed:
test_stats [all random seeds]
test_averaged_weighted_percentile
test_weighted_median_equal_weights
test_weighted_median_integer_weights
test_weighted_percentile_2d

Rishab260 committed:
Merge branch 'main' into test_stats_branch

Rishab260 committed:
Merge remote-tracking branch 'upstream/main' into test_stats_branch

Rishab260 committed:
fix test_weighted_median_integer_weights

Rishab260 committed:
test_stats [all random seeds]
test_averaged_weighted_percentile
test_weighted_median_equal_weights
test_weighted_median_integer_weights
test_weighted_percentile_2d

OmarManzoor commented on sklearn/utils/tests/test_stats.py:
```diff
@@ -98,12 +98,12 @@ def test_weighted_median_equal_weights():
-    x = rng.randint(20, size=10)
-    weights = rng.choice(5, size=10)
+    rng = np.random.RandomState(global_random_seed)
+    x = rng.randint(20, size=1000)
+    weights = rng.choice(5, size=1000)
```
Would size=100 work or does that still fail? I think 1000 is fine anyways.

Rishab260 replied:
Hi, Size 100 failed, we can probably relax the tolerance but since size 1000 worked, I went with that one.

jeremiedbb replied:
The test was not correct. It is not equivalent to np.median when the number of effective samples is even. `_averaged_weighted_percentile` should be equivalent instead. I modified the test accordingly.

OmarManzoor reviewed (approved):
LGTM. Thanks @Rishab260

jeremiedbb committed:
Merge remote-tracking branch 'upstream/main' into pr/Rishab260/30857

jeremiedbb committed:
fix test

jeremiedbb committed:
[all random seeds]
test_averaged_weighted_percentile
test_weighted_median_equal_weights
test_weighted_median_integer_weights
test_weighted_percentile_2d
test_weighted_percentile_array_api_consistency
test_weighted_percentile_nan_filtered
test_weighted_percentile_like_numpy_quantile
test_weighted_percentile_like_numpy_nanquantile

jeremiedbb reviewed (approved):
LGTM. Thanks @Rishab260

jeremiedbb committed:
run full test suite to be safe
